### PR TITLE
bpo-30801: Fix for shoutdown process error with python 3.4 and pyqt/PySide

### PR DESCRIPTION
**Pull request by alfonnews**
This is a fix for an error that it is happend when shutdown manager process with a frozen app.
   https://github.com/pyinstaller/pyinstaller/issues/1937

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by eryksun** [comment]
This has nothing to do with bpo-1754 (WindowsError messages are not properly encoded). Please reference the correct issue.

**Comment by alfonnews** [comment]
I think that I reference the correct issue.
Please correct me if it is not

https://github.com/cython/cython/issues/1754

**Comment by eryksun** [comment]
You created an issue on Cython's issue tracker instead of CPython's issue tracker at [bugs.python.org](http://bugs.python.org).

**Comment by alfonnews** [comment]
Ok, Thanks. I fix them.

**Comment by larryhastings** [comment]
This PR is not targeting the 3.4 branch; it's targeting master.

Python 3.4 is in "security fixes only" mode.  Since this is not a "security fix", it will not be accepted into the 3.4 branch.  Please either change the description so it doesn't mention 3.4, or close the PR.

**Comment by larryhastings** [comment]
If there are no changes to this issue in two weeks, I will close it.

File: Lib/multiprocessing/process.py [hunk 1]
```diff
@@ -296,8 +297,18 @@ def _bootstrap(self):
             traceback.print_exc()
         finally:
             util.info('process exiting with exitcode %d' % exitcode)
-            sys.stdout.flush()
-            sys.stderr.flush()
+            #check if windows platform and Python 3.4
+            if sys.platform.startswith('win') and sys.version_info[:2] == (3, 4):
```
Comment (terryjreedy) [inline Lib/multiprocessing/process.py:301]: The version check makes this useless in any other version.

File: Lib/multiprocessing/process.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import signal
 import itertools
 from _weakrefset import WeakSet
+import imp
```
Comment (terryjreedy) [inline Lib/multiprocessing/process.py:21]: Deprecated in 3.4.  A patch using deprecated modules will not be accepted.

File: Lib/multiprocessing/process.py [hunk 3]
```diff
@@ -296,8 +297,18 @@ def _bootstrap(self):
...
-            sys.stderr.flush()
+            #check if windows platform and Python 3.4
+            if sys.platform.startswith('win') and sys.version_info[:2] == (3, 4):
+                # checks if forzen app
+                is_frozen = (hasattr(sys, "frozen") or # new py2exe
+                             hasattr(sys, "importers") or # old py2exe
+                             imp.is_frozen("__main__")) # tools/freeze
+                if not is_frozen:
+                    sys.stdout.flush()
+                    sys.stderr.flush()
```
Comment (terryjreedy) [inline Lib/multiprocessing/process.py:308]: The fact that one frozen module crashes from the flushes does not mean that all frozen modules will or that suppressing flushes will not harm other frozen modules.  The fact that other frozen modules do run on various versions of Python suggest that the problem is with the particular frozen module.  This suggests to me that the patch is in general wrong and should be rejected now.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I think this should be closed.

**Comment by larryhastings** [comment]
Ouch.  Yeah, this code is awful and would never be merged.

@alfonnews: in the future, if you're interested in contributing to CPython, you should know that we have very high code quality standards.  Your PR here is a "band-aid"; it doesn't fix the problem (whatever it is), it merely works around it.  To be honest I still have no idea what problem you're solving.  And the comments by @terryjreedy are all spot on.

**Comment by alfonnews** [comment]
I agree, in any case sorry for the inconvenience.
My intention was to expose the concrete problem and a casuistic that solves it, not as a final solution but as a starting point to detect the underlying problem and its correct solution.
I understand that it is not the right way for this. Thanks for corrections and suggestions

**Pull request closed without merging**